internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -177,25 +193,57 @@ func (r *Reconciler) reconcile(ctx context.Context, clusterClass *clusterv1.Clus
 		return ctrl.Result{}, kerrors.NewAggregate(errs)
 	}
 
-	// Ensure the variables are added to the ClusterClass status.
+	// Add inline variable definitions to the ClusterClass status.
 	clusterClass.Status.Variables = []clusterv1.ClusterClassStatusVariable{}
 	for _, variable := range clusterClass.Spec.Variables {
-		clusterClass.Status.Variables = append(clusterClass.Status.Variables,
-			clusterv1.ClusterClassStatusVariable{
-				Name: variable.Name,
-				Definitions: []clusterv1.ClusterClassStatusVariableDefinition{
-					{
-						From:     clusterv1.VariableDefinitionFromInline,
-						Required: variable.Required,
-						Schema:   variable.Schema,
-					},
-				}})
+		clusterClass.Status.Variables = append(clusterClass.Status.Variables, statusVariableFromClusterClassVariable(variable, clusterv1.VariableDefinitionFromInline))
+	}
+
+	// If RuntimeSDK is enabled call the DiscoverVariables hook for all associated Runtime Extensions and add the variables
+	// to the ClusterClass status.
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		for _, patch := range clusterClass.Spec.Patches {
+			if patch.External == nil || patch.External.DiscoverVariablesExtension == nil {
+				continue
+			}
+			req := &runtimehooksv1.DiscoverVariablesRequest{}
+			req.Settings = patch.External.Settings
+
+			resp := &runtimehooksv1.DiscoverVariablesResponse{}
+			err := r.RuntimeClient.CallExtension(ctx, runtimehooksv1.DiscoverVariables, clusterClass, *patch.External.DiscoverVariablesExtension, req, resp)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			if resp.Status != runtimehooksv1.ResponseStatusSuccess {
+				return ctrl.Result{}, errors.Errorf("failed to discover variables for ClusterClass %s: %s", clusterClass.Name, resp.Message)
+			}
+			if resp.Variables != nil {
+				for _, variable := range resp.Variables {
+					// TODO: Variables should be vaildated and deduplicated base on their provided definition.
```

Comment:
My plan is to get a first iteration without deduplication, and then do tackle that in a follow up.